### PR TITLE
fix: [L08]: improve executeEmergencyProposal docstring

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@umaproject.org>

**Motivation**

In the last audit OZ signaled:
```
In the EmergencyProposal contract, the executeEmergencyProposal function iterates through the transactions submitted 
on the proposal and executes them. However, only the first element in the array is able to execute a payable transaction,
as the first element of the proposed transactions will transfer the entirety of the contract ETH balance, meaning that 
subsequent transactions will have a value of zero.

Consider documenting this behavior thoroughly as it may not be obvious or intuitive to the proposal submitter.
```

**Summary**

Improve `executeEmergencyProposal` & `emergencyPropose` to clarify how the transactions execution work

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

**Review by mrice32 — COMMENTED** [review]
I'm not sure I agree with this issue. The emergency proposer forwards its entire balance (presumably the amount sent by the caller) to the governor in the first transaction. However, the governor could _spend_ that balance over multiple transactions.

For example, I see no reason why the following wouldn't work:
- Proposer sends 2 ETH with their executeEmergencyProposal transaction. 2 transactions are being executed, both sending 1 ETH.
- EmergencyProposer's forwards the entire 2 ETH on the execution of transaction 1. Governor sends 1 ETH and keeps 1 ETH (since it was sent to the governor, but was not spent).
- EmergencyProposer executes the second transaction. The governor sends its remaining 1 ETH with the transaction.

I think the only requirement is that the execution transaction has the total of all ETH sent in all emergency execution transactions.

**Comment by md0x** [comment]
> 

Thanks @mrice32, I agree with you, I've updated the docstrings accordingly.

**Review by chrismaree — APPROVED** [review]

**Comment by chrismaree** [comment]
disregard my approval, was premature.

**Comment by chrismaree** [comment]
> I'm not sure I agree with this issue. The emergency proposer forwards its entire balance (presumably the amount sent by the caller) to the governor in the first transaction. However, the governor could _spend_ that balance over multiple transactions.
> 
> For example, I see no reason why the following wouldn't work:
> 
> * Proposer sends 2 ETH with their executeEmergencyProposal transaction. 2 transactions are being executed, both sending 1 ETH.
> * EmergencyProposer's forwards the entire 2 ETH on the execution of transaction 1. Governor sends 1 ETH and keeps 1 ETH (since it was sent to the governor, but was not spent).
> * EmergencyProposer executes the second transaction. The governor sends its remaining 1 ETH with the transaction.
> 
> I think the only requirement is that the execution transaction has the total of all ETH sent in all emergency execution transactions.
> 
> @md0x @chrismaree do y'all agree?

I agree. as long as the sum of all governor transactions msg.value = the msg.value sent in the tx from the proposer you can split multiple payable functions calls from the governor.

File: packages/core/contracts/oracle/implementation/EmergencyProposer.sol [hunk 1]
```diff
@@ -171,6 +171,9 @@ contract EmergencyProposer is Ownable, Lockable {
     /**
      * @notice After a proposal expires, this method can be used by the executor to execute the proposal.
      * @dev This method effectively gives the executor veto power over any proposal.
+     * @dev The first transaction execution sends the total amount of ETH required to complete all payable
+     * transactions in the Governor. The EmergencyProposer must receive this amount of ETH in advance.
+     * The Governor can then decide how each of the subsequent transactions will use this ETH.
```
Comment (mrice32) [inline packages/core/contracts/oracle/implementation/EmergencyProposer.sol:176]:
```suggestion
     * The executed transactions are then able to use this ETH by including a nonzero value.
```

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**